src/libs/SearchUIUtils.ts

```diff
@@ -0,0 +1,524 @@
+    [CONST.SEARCH.TABLE_COLUMNS.TYPE]: 'transactionType' as const,
+    [CONST.SEARCH.TABLE_COLUMNS.ACTION]: 'action' as const,
+    [CONST.SEARCH.TABLE_COLUMNS.DESCRIPTION]: 'comment' as const,
+    [CONST.SEARCH.TABLE_COLUMNS.TAX_AMOUNT]: null,
+    [CONST.SEARCH.TABLE_COLUMNS.RECEIPT]: null,
+};
+
+const emptyPersonalDetails = {
+    accountID: CONST.REPORT.OWNER_ACCOUNT_ID_FAKE,
+    avatar: '',
+    displayName: undefined,
+    login: undefined,
+};
+
+type ReportKey = `${typeof ONYXKEYS.COLLECTION.REPORT}${string}`;
+
+type TransactionKey = `${typeof ONYXKEYS.COLLECTION.TRANSACTION}${string}`;
+
+type ReportActionKey = `${typeof ONYXKEYS.COLLECTION.REPORT_ACTIONS}${string}`;
+
+/**
+ * @private
+ *
+ * Returns a list of properties that are common to every Search ListItem
+ */
+function getTransactionItemCommonFormattedProperties(
+    transactionItem: SearchTransaction,
+    from: SearchPersonalDetails,
+    to: SearchPersonalDetails,
+): Pick<TransactionListItemType, 'formattedFrom' | 'formattedTo' | 'formattedTotal' | 'formattedMerchant' | 'date'> {
+    const isExpenseReport = transactionItem.reportType === CONST.REPORT.TYPE.EXPENSE;
+
+    const formattedFrom = from?.displayName ?? from?.login ?? '';
+    const formattedTo = to?.displayName ?? to?.login ?? '';
+    const formattedTotal = TransactionUtils.getAmount(transactionItem, isExpenseReport);
+    const date = transactionItem?.modifiedCreated ? transactionItem.modifiedCreated : transactionItem?.created;
+    const merchant = TransactionUtils.getMerchant(transactionItem);
+    const formattedMerchant = merchant === CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT || merchant === CONST.TRANSACTION.DEFAULT_MERCHANT ? '' : merchant;
+
+    return {
+        formattedFrom,
+        formattedTo,
+        date,
+        formattedTotal,
+        formattedMerchant,
+    };
+}
+
+/**
+ * @private
+ */
+function isReportEntry(key: string): key is ReportKey {
+    return key.startsWith(ONYXKEYS.COLLECTION.REPORT);
+}
+
+/**
+ * @private
+ */
+function isTransactionEntry(key: string): key is TransactionKey {
+    return key.startsWith(ONYXKEYS.COLLECTION.TRANSACTION);
+}
+
+/**
+ * @private
+ */
+function isReportActionEntry(key: string): key is ReportActionKey {
+    return key.startsWith(ONYXKEYS.COLLECTION.REPORT_ACTIONS);
+}
+
+/**
+ * Determines whether to display the merchant field based on the transactions in the search results.
+ */
+function getShouldShowMerchant(data: OnyxTypes.SearchResults['data']): boolean {
+    return Object.keys(data).some((key) => {
+        if (isTransactionEntry(key)) {
+            const item = data[key];
+            const merchant = item.modifiedMerchant ? item.modifiedMerchant : item.merchant ?? '';
+            return merchant !== '' && merchant !== CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT && merchant !== CONST.TRANSACTION.DEFAULT_MERCHANT;
+        }
+        return false;
+    });
+}
+
+const currentYear = new Date().getFullYear();
```

Comment:
NAB this is not your code, but it's kinda odd that we have this random const here when it's only used within shouldShowYear

Reply:
Good point, will fix